### ISSUE TITLE
Asset lib or Templates are not accessible on the hosts with disabled ipv6.

### ISSUE THREAD

**Issue by syskrank** [godotengine/godot#7676]
**Operating system or device - Godot version:**
GNU/Linux Debian 8.6 jessie
Godot: 2.1.2 stable/custom build

**Issue description:**
Asset lib from within the editor and Templates from project list are not accessible on the hosts with disabled ipv6 ( "Can't connect to host:  https://godotengine.org/asset-library/api" )

**Steps to reproduce:**
1. disable ipv6 on your machine
2. launch godot editor with -d flag
3. try to open 'templates' tab in the project selector
OR
open any project for editing and navigate to asset lib
4. see the "Can't connect to host: https://godotengine.org/asset-library/api

and the corresponding console log:
```
ERROR: _socket_create: Condition ' sockfd == -1 ' is true. returned: -1
   At: ./drivers/unix/socket_helpers.h:73.
ERROR: connect: Socket creation failed!
   At: drivers/unix/stream_peer_tcp_posix.cpp:148.
ERROR: _socket_create: Condition ' sockfd == -1 ' is true. returned: -1
   At: ./drivers/unix/socket_helpers.h:73.
ERROR: connect: Socket creation failed!
   At: drivers/unix/stream_peer_tcp_posix.cpp:148.
```

**Comment by Faless** [comment]
This is already fixed in master via #7510, but was not cherry picked in time for 2.1 .
You can try [this patch](https://github.com/Faless/godot/commit/43bc27e2755f34017148097773da8a6988718aaf.diff) . (Just cherry picked 55b4f3686d8f51958132a1a1745cc4e128fd118d from master)

**Comment by syskrank** [comment]
Thanks for the tip. Didn't know that this issue is already fixed in 3.0 so I managed to fix this myself. I created a [pull request](https://github.com/godotengine/godot/pull/7677) for 2.1 recently. Introduced code adds some verbosity via strerror() and utilizes a simple socket domain checker to verify that ipv6 is available in the system.

**Comment by akien-mga** [comment]
Fixed by #8108.
